libs/ha/ha_common_libs_k8s.py

```diff
@@ -1442,47 +1444,67 @@ def get_node_resource_ids(node_obj, r_type, n_type=None, node_id=None):
         return resp
 
     def delete_single_pod_setting_replica_0(self, master_node_obj, health_obj,
-                                            pod_prefix=common_const.POD_NAME_PREFIX):
+                                            pod_prefix=common_const.POD_NAME_PREFIX,
+                                            down_method=common_const.RESTORE_SCALE_REPLICAS):
         """
         Delete Single pod by setting replica set to 0. Check service status is in degraded mode.
         :param master_node_obj: Master node object list
         :param health_obj: Health object
         :param pod_prefix: Pod prefix to be deleted.
+        :param down_method: Pod shutdown/delete method.
         return : tuple
         """
+
+        down_info = {"method": None, "deployment_name": None, "deployment_backup": None}
+        remaining = list()
         LOGGER.info("Get pod name to be deleted")
         pod_list = master_node_obj.get_all_pods(pod_prefix=pod_prefix)
         pod_name = random.sample(pod_list, 1)[0]
         hostname = master_node_obj.get_pod_hostname(pod_name=pod_name)
-
-        LOGGER.info("Deleting pod %s", pod_name)
-        resp = master_node_obj.create_pod_replicas(num_replica=0, pod_name=pod_name)
-        if resp[0]:
-            return False, f"Failed to delete pod {pod_name} by making replicas=0"
-        LOGGER.info("Successfully shutdown/deleted pod %s by making replicas=0", pod_name)
+        total_type = [common_const.POD_NAME_PREFIX, common_const.SERVER_POD_NAME_PREFIX]
+        total_type.remove(pod_prefix)
```

Comment:
Why pod_prefix is removed?
Instead we can take pod_prefix_list as user input.

Reply:
> Why pod_prefix is removed? Instead we can take pod_prefix_list as user input. It will be useful in case of both server and data pod deletion.

Done